File: T1.SqlSharp/T1.SqlSharp/Expressions/SqlComparisonOperator.cs
namespace T1.SqlSharp.Expressions;

public class SqlComparisonOperator : ISqlExpression
{
    public SqlType SqlType { get; } = SqlType.ComparisonOperator;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_ComparisonOperator(this);
    }

    public ComparisonOperator Value { get; set; }
    public string ToSql()
    {
        return Value.ToSql();
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlComparisonOperator.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';
import { ComparisonOperator } from './ComparisonOperator';
import { comparisonOperatorToSql } from './ComparisonOperatorExtensions';

export class SqlComparisonOperator implements ISqlExpression {
    SqlType: SqlType = SqlType.ComparisonOperator;
    Span: TextSpan = new TextSpan();
    Value!: ComparisonOperator;

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_ComparisonOperator(this);
    }

    ToSql(): string {
        return comparisonOperatorToSql(this.Value);
    }
} 